user_sync/rules.py

```diff
@@ -568,11 +622,26 @@ def update_dashboard_users_for_connector(self, organization_info, dashboard_conn
                     self.logger.info('Updating info for user key: %s changes: %s', user_key, user_attribute_difference)
             
             groups_to_add = None
-            groups_to_remove = None    
-            if (manage_groups):        
-                current_groups = self.normalize_groups(dashboard_user.get('groups'))
-                groups_to_add = desired_groups - current_groups 
-                groups_to_remove =  (current_groups - desired_groups) & organization_info.get_mapped_groups()                
+            groups_to_remove = None
+            if (manage_groups):
+                # generate a list of current groups to determine what to delete
+                current_group_names = self.normalize_groups(dashboard_user.get('groups'))
+                current_groups = set()
+                for current_group_name in current_group_names:
+                    current_groups.add(TargetGroup(current_group_name))
```

Comment:
I allowed it to create it this way to make set math easier, I've added functions to allow filtering by group names...